### PR TITLE
[docs] Add an example of doing `str = readline()` to read a single line in from `stdin`

### PR DESCRIPTION
**Pull request by logankilpatrick**
I would assume that this might lead to doctest issues but having an example of the io readline interface seems like it is something we should do so I added this simple example.

### CODE REVIEW

**Comment by DilumAluthge** [comment]
If we can't figure out how to get a doctest to work for this, you could make it a separate non-doctest example?

**Comment by logankilpatrick** [comment]
Yeah, not sure if this will fail the doc test though, either way, it should be added.

**Comment by DilumAluthge** [comment]
The CI doctest job seems to be falling.

**Comment by DilumAluthge** [comment]
On a separate note, would it be better to have two examples? One for `your_name = readline()` and one for `your_name = readline(stdlin)`? Or would that be excessive?

**Comment by logankilpatrick** [comment]
Is there a benefit too using `readline(stdin)` over `readline()`? If not, perhaps the simpler version is all that is needed.

**Comment by DilumAluthge** [comment]
No, it would just be to illustrate how people can pass in their own streams instead of using standard input.

I agree that the simpler version is probably all we need.

**Comment by IanButterworth** [comment]
Perhaps it's odd to direct people to construct this themselves while `Base.prompt` exists?

**Comment by DilumAluthge** [comment]
Is `Base.prompt` public? I can't seem to find it in the manual.

**Comment by DilumAluthge** [comment]
If `Base.prompt` is indeed intended to be part of the public API, I would suggest that we:
1. Add the `Base.prompt` docstring to a manual page.
2. Add a brief usage example (similar to Logan's example in this PR) to the `Base.prompt` docstring (https://github.com/JuliaLang/julia/blob/f8f42ec0370b72072498c1b79eb5cf4e6c6864e6/base/util.jl#L288)
3. Close this PR.

**Comment by DilumAluthge** [comment]
> Is `Base.prompt` part of the public API? I can't seem to find it in the manual.

cc @omus, since it looks like you moved `prompt` from LibGit2 to Base in https://github.com/JuliaLang/julia/pull/23092/commits/f26ee56924e914d2ac80dbdd550d9c861c3768b2 (part of https://github.com/JuliaLang/julia/pull/23092).

**Comment by logankilpatrick** [comment]
I would suggest having both, every online site points to readline for user input so it's odd we don't document it here.

**Comment by DilumAluthge** [comment]
> Add the `Base.prompt` docstring to a manual page.

https://github.com/JuliaLang/julia/pull/43637

**Comment by DilumAluthge** [comment]
> Add a brief usage example (similar to Logan's example in this PR) to the `Base.prompt` docstring

https://github.com/JuliaLang/julia/pull/43638

File: base/io.jl [hunk 1]
```diff
@@ -511,6 +511,13 @@ line.
 
 # Examples
 ```jldoctest
+julia> print("Enter your name: ")
+Enter your name:
+
```
Comment (DilumAluthge) [inline base/io.jl:516]: Required in order to satisfy the whitespace check:

```suggestion

```

**Comment by DilumAluthge** [comment]
Yeah, CI is not happy with this as a doctest. Can you make a separate `julia-repl` code block for this example?

File: base/io.jl [hunk 2]
```diff
@@ -510,6 +510,14 @@ line before it is returned. When `keep` is true, they are returned as part of th
 line.
 
 # Examples
+```julia-repl
+julia> print("Enter your name: ")
+Enter your name:
+
+julia> your_name = readline()
+Logan
+"Logan"
```
Comment (DilumAluthge) [inline base/io.jl:519]:
```suggestion
"Logan"
```

File: base/io.jl [hunk 3]
```diff
@@ -510,6 +510,14 @@ line before it is returned. When `keep` is true, they are returned as part of th
...
 
 # Examples
+```julia-repl
+julia> print("Enter your name: ")
+Enter your name:
+
+julia> your_name = readline()
+Logan
+"Logan"  
+```
```
Comment (DilumAluthge) [inline base/io.jl:520]:
````suggestion
```
````

File: base/io.jl [hunk 4]
```diff
@@ -510,6 +510,14 @@ line before it is returned. When `keep` is true, they are returned as part of th
 line.
 
 # Examples
+```julia-repl
```
Comment (DilumAluthge) [inline base/io.jl:513]: Does this example need to go first? I would probably recommend putting it below the existing examples, unless it really needs to be at the top.

**Review by DilumAluthge — APPROVED** [review]

**Comment by omus** [comment]
> > Is `Base.prompt` part of the public API? I can't seem to find it in the manual.
> 
> cc @omus, since it looks like you moved `prompt` from LibGit2 to Base in [f26ee56](https://github.com/JuliaLang/julia/commit/f26ee56924e914d2ac80dbdd550d9c861c3768b2) (part of #23092).

It seems reasonable to me to add `Base.prompt` to the public API by adding it into the manual: #43637

**Pull request merged**